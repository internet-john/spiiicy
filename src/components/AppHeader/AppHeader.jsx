import React from "react";
import { StyleSheet } from "react-native";
import { Header } from "react-native-elements";

const AppHeader = () => (
  <Header
    centerComponent={{ text: "Â¡spiiicy!", style: styles.headerText }}
    containerStyle={styles.header}
  />
);

const styles = StyleSheet.create({
  header: {
    backgroundColor: "#101010",
    borderBottomWidth: 0,
    marginTop: 50,
    marginBottom: 10,
  },
  headerText: {
    color: "red",
    fontSize: 32,
    fontWeight: "800",
    letterSpacing: -2.5,
  },
});

export default AppHeader;
