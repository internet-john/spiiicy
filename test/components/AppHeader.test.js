import React from "react";
import renderer from "react-test-renderer";
import AppHeader from "../../src/components/AppHeader";

test("AppHeader renders as expected", () => {
  const header = renderer.create(<AppHeader />).toJSON();

  expect(header).toMatchSnapshot();
});
