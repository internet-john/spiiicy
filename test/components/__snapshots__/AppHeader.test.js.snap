// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#101010",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "marginTop": 50,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  testID="headerContainer"
>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "top",
        "right",
      ]
    }
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "red",
            "fontSize": 32,
            "fontWeight": "800",
            "letterSpacing": -2.5,
          }
        }
        text="¡spiiicy!"
      >
        ¡spiiicy!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </RNCSafeAreaView>
</View>
`;
